apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/WalletOverview/WalletOverview.tsx

```diff
@@ -1,45 +1,44 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 
 import { t } from 'i18next';
 
+import { AssetDetailsData, getAssetData } from '@sovryn/contracts';
 import {
   Button,
   ErrorBadge,
   ErrorLevel,
   Paragraph,
   ParagraphSize,
-  Tabs,
-  TabSize,
-  TabType,
 } from '@sovryn/ui';
 import { Decimal } from '@sovryn/utils';
 
+import { BOB_CHAIN_ID } from '../../../../../config/chains';
+
 import { AmountRenderer } from '../../../../2_molecules/AmountRenderer/AmountRenderer';
-import { ETH, WETH } from '../../../../../constants/currencies';
 import { useWalletConnect } from '../../../../../hooks';
+import { useAaveUserReservesData } from '../../../../../hooks/aave/useAaveUserReservesData';
 import { translations } from '../../../../../locales/i18n';
 import { BorrowAction } from './components/BorrowAction/BorrowAction';
 import { SupplyAction } from './components/SupplyAction/SupplyAction';
 
 const pageTranslations = translations.aaveReserveOverviewPage;
 
-const ETH_ASSET_SYMBOLS = [ETH, WETH];
-
 type WalletOverviewProps = {
-  asset: {
-    symbol: string;
-    name: string;
-  };
+  symbol: string;
 };
 
-export const WalletOverview: FC<WalletOverviewProps> = ({
-  asset: initialAsset,
-}) => {
-  const [asset, setAsset] = useState(initialAsset.symbol);
+export const WalletOverview: FC<WalletOverviewProps> = ({ symbol }) => {
+  const [asset, setAsset] = useState<AssetDetailsData | null>();
   const { account, connectWallet, pending } = useWalletConnect();
+  const { summary } = useAaveUserReservesData();
```

Comment:
You can move this up to `AaveReserveOverviewPage` to avoid double fetching, also enforce it there allowing you to remove undefined checks from here